src/main/java/se/liu/ida/hefquin/engine/wrappers/graphqlwrapper/impl/GraphQLEndpointInitializerImpl.java

```diff
@@ -151,6 +152,67 @@ protected static GraphQLQuery getIntrospectionQuery(){
         return new GraphQLQueryImpl(fieldPaths, new HashSet<>());
     }
 
+    /**
+     * Removes the types, fields and entrypoints from the @param typesAndFields and @param entrypoints
+     * and notes their removal if they don't meet the criteria for the approach. (each type needs an id field etc.)
+     */
+    protected GraphQLEndpoint validateTypesAndEntrypoints(final Map<String, Map<String, GraphQLField>> typesAndFields,
+            final Map<String, Map<GraphQLEntrypointType,GraphQLEntrypoint>> entrypoints,
+            final GraphQLInterface iface){
+
+        final Set<String> typesToRemove = new HashSet<>();
+
+        // Check if any types needs to be removed
+        for(final String type : typesAndFields.keySet()){
+            final Map<String,GraphQLField> fields = typesAndFields.get(type);
+
+            // Check if type has an id field
+            if(!fields.containsKey("id")){
+                System.out.println("Type: " + type + " is removed due to not having an id field!");
```

Comment:
Also here, are these prints really necessary or were useful for you while developing?

Reply:
They are not necessary. They are now removed in latest commit: [94c8ce5](https://github.com/LiUSemWeb/HeFQUIN/pull/237/commits/94c8ce59988204af9c00434a7f9acc14d5c8e053)